LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DownlinkMessageBuilder.cs

```diff
@@ -77,7 +77,12 @@ internal static class DownlinkMessageBuilder
             else
             {
                 datr = loRaRegion.GetDownstreamDR(rxpk, (uint)loRaDevice.ReportedRX1DROffset);
-                freq = loRaRegion.GetDownstreamChannelFrequency(rxpk);
+                if (!loRaRegion.TryGetDownstreamChannelFrequency(rxpk, out freq) && datr != null)
```

Comment:
Here will continue if we cannot get a frequency and datr. Shouldn't it stop if any of those fail (or)?

Reply:
I am not sure I understand, I am returning an empty object that should be ignored by the receiver. What changes do you intend?

Reply:
I think the intend is to stop processing if we could not find a datr or frequency. It looks to me that the code will only stop if we cannot find a frequency **but found a datr**.

Reply:
I think the logic is indeed wrong, I changed the && to a or